romancal/associations/tests/test_skycell_asn.py

```diff
@@ -30,3 +31,180 @@ def test_parse_visitID():
     assert visitid_parts["Segment"] == "003"
     assert visitid_parts["Observation"] == "004"
     assert visitid_parts["Visit"] == "005"
+
+
+@pytest.fixture
+def sample_filelist():
+    return [
+        "r0000101002003004005_0001_wfi10_cal.asdf",
+        "r0000101002003004005_0002_wfi10_cal.asdf",
+        "r0000101002003004006_0001_wfi10_cal.asdf",
+    ]
+
+
+def test_create_groups_full(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "full")
+    assert "full" in groups
+    assert set(groups["full"]) == set(sample_filelist)
+
+
+def test_create_groups_visit(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "visit")
+    # Should group by visit id (first 19 chars after 'r')
+    assert all(isinstance(v, list) for v in groups.values())
+    assert len(groups) == 2  # two unique visit IDs in sample_filelist
+
+
+def test_create_groups_pass(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "pass")
+    assert all(isinstance(v, list) for v in groups.values())
+    # Should group by pass key (positions 5:10)
+    assert set(groups.keys())  # keys should not be empty
+
+
+def test_create_intersecting_skycell_index(monkeypatch, sample_filelist):
+    class DummyCalFile:
+        class Meta:
+            class Instrument:
+                optical_element = "F158"
+
+            instrument = Instrument()
+
+            class WCS:
+                pass
+
+            wcs = WCS()
+
+        meta = Meta()
+
+        def close(self):
+            pass
+
+    monkeypatch.setattr(skycell_asn.rdm, "open", lambda fname: DummyCalFile())
+    monkeypatch.setattr(skycell_asn.sm, "find_skycell_matches", lambda wcs: [1, 2])
+    file_index = skycell_asn._create_intersecting_skycell_index(sample_filelist)
+    assert len(file_index) == len(sample_filelist)
+    for rec in file_index:
+        assert rec[2] == "f158" or rec[2] == "F158".lower()
```

Comment:
I can't figure out the difference between f158 and F158.lower()

Reply:
None. ;)
Fixed it. Thanks for the catch!